### PR TITLE
cilium: Update to 1.10.4

### PR DESCRIPTION
**Pull request by vadorovsky**
Update Cilium from 1.9.8 to 1.10.4. Changelog:

https://github.com/cilium/cilium/releases/tag/v1.10.4

Ref: rancher/rke2#1615

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

### CODE REVIEW

File: packages/cilium/package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-url: https://helm.cilium.io/cilium-1.9.8.tgz
-packageVersion: 08
+url: https://helm.cilium.io/cilium-1.10.4.tgz
+packageVersion: 09
```
Comment (manuelbuil) [inline packages/cilium/package.yaml:2]: If the base tarball changes, we can start from 01 again

File: packages/rke2-cilium/package.yaml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 url: local
-packageVersion: 09
+packageVersion: 10
```
Comment (manuelbuil) [inline packages/rke2-cilium/package.yaml:2]: Same

File: packages/cilium/package.yaml (hunk 1 above)
Comment (vadorovsky) [reply packages/cilium/package.yaml]: done

File: packages/rke2-cilium/package.yaml (hunk 2 above)
Comment (vadorovsky) [reply packages/rke2-cilium/package.yaml]: done

**Review by manuelbuil — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**